components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/SubscriptionDataHolder.java

```diff
@@ -65,16 +65,19 @@ public void unregisterTenantSubscriptionStore(String tenantDomain) {
     }
 
     public SubscriptionDataStore getTenantSubscriptionStore(String tenantDomain) {
-        SubscriptionDataStore subscriptionDataStore = subscriptionStore.get(tenantDomain);
-        if (subscriptionDataStore == null) {
-            synchronized (tenantDomain.concat("getTenantSubscriptionStore").intern()) {
-                subscriptionDataStore = subscriptionStore.get(tenantDomain);
-                if (subscriptionDataStore == null) {
-                    subscriptionDataStore = registerTenantSubscriptionStore(tenantDomain);
+        if (subscriptionStore != null && tenantDomain != null) {
+            SubscriptionDataStore subscriptionDataStore = subscriptionStore.get(tenantDomain);
+            if (subscriptionDataStore == null) {
+                synchronized (tenantDomain.concat("getTenantSubscriptionStore").intern()) {
+                    subscriptionDataStore = subscriptionStore.get(tenantDomain);
+                    if (subscriptionDataStore == null) {
+                        subscriptionDataStore = registerTenantSubscriptionStore(tenantDomain);
+                    }
                 }
             }
+            return subscriptionDataStore;
         }
-        return subscriptionDataStore;
+        return null;
```

Comment:
Did we check if there are any possible null pointers that can occur as a result of sending null.

Reply:
When analyzing the master branch, line 80 shouldn't get hit. This `null` check is added in https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/AbstractKeyValidationHandler.java#L219